src/dsl.jl

```diff
@@ -358,17 +358,27 @@ function make_reaction_system(ex::Expr; name = :(gensym(:ReactionSystem)))
 
     # Collect species and parameters, including ones inferred from the reactions.
     declared_syms = Set(Iterators.flatten((parameters_declared, species_declared,
-        variables)))
```

Comment:
The bit of code that generates variables is now  moved later (as it need to know the inferred species). Hence we only have (and only need) `variables_declared`.